zmon-controller-app/src/main/resources/db/migration/R__function_create_or_update_check_definition.sql

```diff
@@ -0,0 +1,145 @@
+
+    -- Find ID and previous runtime of the check and then lock it for update if exists.
+    SELECT cd_id, COALESCE(cd_runtime, 'PYTHON_2')::zzm_data.definition_runtime
+    INTO entity.id, previous_runtime
+    FROM zzm_data.check_definition
+    WHERE (lower(cd_source_url) = lower(check_definition_import.source_url) AND check_definition_import.id IS NULL)
+       OR (lower(cd_name) = lower(check_definition_import.name) AND lower(cd_owning_team) = lower(check_definition_import.owning_team) AND check_definition_import.id IS NULL)
+       OR (cd_id = check_definition_import.id)
+    FOR UPDATE;
+
+    IF FOUND THEN
+        -- Get initial runtime of the check in order to...
+        SELECT COALESCE(cdh_row_data->'cd_runtime', 'PYTHON_2')::zzm_data.definition_runtime
+        INTO initial_runtime
+        FROM zzm_data.check_definition_history
+        WHERE cdh_check_definition_id = entity.id AND cdh_action = 'INSERT'::zzm_data.history_action;
+
+        -- Disallow changing runtime at all if it is globally disabled
+        IF NOT runtime_enabled AND check_definition_import.runtime != previous_runtime THEN
+            permission_denied = true;
+            RETURN;
+        END IF;
+        -- Disallow switching runtime back to Python 2 if a check was created with Python 3 initially.
+        IF runtime_enabled AND initial_runtime = 'PYTHON_3' AND check_definition_import.runtime = 'PYTHON_2' THEN
+            permission_denied = true;
+            RETURN;
+        END IF;
+
+        UPDATE zzm_data.check_definition
+        SET cd_name                 = check_definition_import.name,
+            cd_description          = check_definition_import.description,
+            cd_technical_details    = check_definition_import.technical_details,
+            cd_potential_analysis   = check_definition_import.potential_analysis,
+            cd_potential_impact     = check_definition_import.potential_impact,
+            cd_potential_solution   = check_definition_import.potential_solution,
+            cd_owning_team          = check_definition_import.owning_team,
+            cd_entities             = check_definition_import.entities,
+            cd_interval             = check_definition_import.interval,
+            cd_command              = check_definition_import.command,
+            cd_status               = check_definition_import.status,
+            cd_source_url           = check_definition_import.source_url,
+            cd_last_modified_by     = check_definition_import.last_modified_by,
+            cd_last_modified        = now(),
+            cd_runtime              = check_definition_import.runtime
+        WHERE cd_id = entity.id
+        RETURNING cd_id INTO entity.id;
+    ELSIF NOT FOUND AND check_definition_import.id IS NULL THEN
+        -- Disallow of creating new checks with Python 2.
+        IF runtime_enabled AND check_definition_import.runtime != 'PYTHON_3' THEN
```

Comment:
minor, but sounds like this should be `== 'PYTHON_2'`, as that's what the comment above says this code does

Reply:
Yes, you're right. I am not sure what happened here :D